src/main/java/redis/clients/jedis/JedisClusterCommand.java

```diff
@@ -79,62 +79,87 @@ public T runWithAnyNode() {
     }
   }
 
-  private T runWithRetries(final int slot, int attempts, boolean tryRandomNode, JedisRedirectionException redirect) {
-    if (attempts <= 0) {
-      throw new JedisClusterMaxAttemptsException("No more cluster attempts left.");
-    }
+  private interface ConnectionGetter {
+    Jedis getConnection();
+  }
 
-    Jedis connection = null;
-    try {
+  private T runWithRetries(final int slot) {
+    ConnectionGetter connectionGetter = new ConnectionGetter() {
+      @Override
+      public Jedis getConnection() {
+        return connectionHandler.getConnectionFromSlot(slot);
+      }
+    };
```

Comment:
Repeated code (i).

Reply:
The `ConnectionGetter` uses the `slot`, so I don't think there can be a default one.

With Java 8 lambdas this code would be shorter at least, but currently the source code is Java 7 so that's a no go.

Reply:
Also, a `Thread.sleep()` is likely to go into the other one soon, so the code won't stay duplicated for long.

Reply:
> The ConnectionGetter uses the slot, so I don't think there can be a default one.

Can't you take a slot parameter?

> a Thread.sleep() is likely to go into the other one

In that case, you should make it configurable, meaning user can choose to use sleep or not. It should be `not` by default as this is the current behavior.

Reply:
We applied the changes from #2357 use lambdas instead of ConnectionGetters now.

I don't think `() -> connectionHandler.getConnectionFromSlot(slot)` needs to be de-duplicated, what do you think @sazzad16?

Reply:
We'll look into it later. Don't worry about it for now.